src/main-process/atom-window.js

```diff
@@ -464,13 +464,13 @@ module.exports = class AtomWindow extends EventEmitter {
       options
     );
 
+    let promise = dialog.showSaveDialog(this.browserWindow, options);
     if (typeof callback === 'function') {
-      // Async
-      dialog.showSaveDialog(this.browserWindow, options, callback);
-    } else {
-      // Sync
-      return dialog.showSaveDialog(this.browserWindow, options);
+      promise = promise.then(({ filePath, bookmark }) => {
+        callback(filePath, bookmark);
+      });
```

Comment:
For the case when there is a callback to run, maybe we should log a deprecation notice. (Although I would think it would be good to have feedback from Atom's official maintainers for that?)

Reply:
This implementation, could also serve a double purposes if we so something like this:
```js
      promise = promise.then((data) => {
        callback(data.filePath, data.bookmark);
        return data;
      });
```
In this way it would allow for using the method as a promise.
But in that case would need to add a test.

Reply:
It is beyond my skill level with Promises to really comment on that, but it sounds reasonable.

Reply:
It's better to have it serve one purpose. Looking at ` showSaveDialog` usage we don't gain anything by making it be used as a promise.